main.go

```diff
@@ -2,16 +2,101 @@ package main
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"runtime"
 	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/urfave/cli/v2"
+	"github.com/vshn/go-bootstrap/cfg"
+	"github.com/vshn/go-bootstrap/cmd"
+	"github.com/vshn/go-bootstrap/cmd/example"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 var (
 	// these variables are populated by Goreleaser when releasing
 	version = "unknown"
 	commit  = "-dirty-"
 	date    = time.Now().Format("2006-01-02")
+
+	// TODO: Adjust app name
+	appName     = "go-bootstrap"
+	appLongName = "a generic bootstrapping project"
 )
 
 func main() {
-	fmt.Printf("version %s, commit %s, date %s\n", version, commit, date)
+	err := app().Run(os.Args)
+	if err != nil {
+		log.Fatalf("unable to start %s: %v", appName, err)
+	}
+}
+
+func before(c *cli.Context) error {
+	logger := newLogger(appName, c.Bool("debug"))
+	cmd.SetAppLogger(c, logger)
```

Comment:
I'd also call [`zap.ReplaceGlobals()`](https://pkg.go.dev/go.uber.org/zap#ReplaceGlobals) for consistency.

With the global zap logger I'n not sure if we really need to save the logger to the command context.

Reply:
And here's me that doesn't want to use zap's global logger. This would make a hard dependency to zap, which I'm trying to avoid by using the interface-only `logr` abstraction.

Reply:
> And here's me that doesn't want to use zap's global logger

Fair enough. I see the use-case of the `cmd.SetAppLogger`

I'd still do the `ReplaceGlobals` if someone decides to use the logger. It won't introduce anymore dependency than there is already.